joshua/joshua_agent.py

```diff
@@ -519,6 +522,28 @@ def _run_ensemble(
 
         #    print('{} Running ensemble in dir: {}'.format(threading.current_thread().name, work_dir),file=getFileHandle())
 
+        # Check and set the Pod label when run inside a k8s pod
+        # env['HOSTNAME'] contains the pod name in k8s
+        namespace = ""
+        k8s_namespace_file = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+        if env['HOSTNAME'] and os.path.isfile(k8s_namespace_file):
+            pod_name = env['HOSTNAME']
+            with open(k8s_namespace_file, 'r') as f:
+                namespace = f.read()
+
+            # Get the cluster config
+            config.load_incluster_config()
+            v1 = client.CoreV1Api()
+
+            # If last_test label is set to true, exit right away
+            pod_desc = v1.read_namespaced_pod(pod_name, namespace)
+            if pod_desc.metadata.labels.get('last_test', 'false') == "true":
```

Comment:
```suggestion
            if pod_desc.metadata.labels['last_test']:
```

With the assumption that `last_test` will always be true if set we could simplify it like this, but i know to little about the setup :)

I haven't worked with the Python Kubernetes client for a while, but could there be a case where a Pod doesn't have labels? If we are sure our Pods always have at lease one label we can ignore that case.